reviewing/plugin-commands-outdated/test/index.ts

```diff
@@ -17,6 +17,7 @@ const hasNotOutdatedDepsFixture = f.find('has-not-outdated-deps')
 const hasMajorOutdatedDepsFixture = f.find('has-major-outdated-deps')
 const hasNoLockfileFixture = f.find('has-no-lockfile')
 const withPnpmUpdateIgnore = f.find('with-pnpm-update-ignore')
+const hasOutdatedDepsUsingCatalogProtocol = f.find('has-outdated-deps-using-catalog-protocol')
```

Comment:
I think we should use `preparePackage` for these tests?

I setup an in-repo fixture since the other tests in `plugin-commands-outdated` followed that structure, but I personally prefer programmatically creating the test fixture in a temp dir, which `preparePackage` does.

Reply:
I have no objections to dynamically generate the fixture with preparePackage

Reply:
I'm realizing why the `plugin-commands-outdated` tests all rely on fixtures. After I tried to use `preparePackage`, I hit this error:

<img width="842" alt="Screenshot 2024-07-15 at 1 19 04 AM" src="https://github.com/user-attachments/assets/fa58b2a0-9f6a-4cef-a476-c433c64e3da9">

To create a `pnpm-lock.yaml`, we could import `@pnpm/core` here and run `mutateModules`, but I don't see any other tests in the pnpm repo doing that other than the tests in `@pnpm/core` itself. We'd also need to copy `pkg-manager/core/test/utils/testDefaults.ts`.

I could see that being valuable in general though. Perhaps a new `@pnpm/test-defaults` package might be useful.